File: src/main/java/GuessNumberHigherOrLower/wally/GuessNumberHigherOrLower.java
package GuessNumberHigherOrLower.wally;

/**
 * @author wally
 */
public class GuessNumberHigherOrLower {
    /**
     * 使用二分法进行遍历。
     * @param n
     * @return
     */
    public int guessNumber(int n){
        int left = 1;
        int right = n;
        for (int i = 0; i <= n /2 ;i++){
            int mid = left + (right - left) / 2;
            if (0 == guess(mid)){
                System.out.println("Congrats! You got it!");
                return mid;
            }else if (-1 == guess(mid)){
                right = mid - 1;
                System.out.println("My number is lower");
            }else if(1 == guess(mid)){
                left = mid + 1;
                System.out.println("My number is higher");
            }
        }
        return 0;
    }
    public static int guess(int n){
        return n;
    }
}

File: src/main/java/GuessNumberHigherOrLower/zhanwenta/GuessNumberHigherOrLower.py
#!/usr/bin/python
# -*- coding: UTF-8 -*-
#!@Author:zhangwentao

# The guess API is already defined for you.
# @param num, your guess
# @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
# def guess(num):

class Solution(object):
    def guessNumber(self, n):
        """
        :type n: int
        :rtype: int
        """
        return self.guessNumberExe(1,n)

    def guessNumberExe(self,head,end):
        if head == end : return head
        zj = (head + end) / 2
        result = guess(zj)
        if result == 1:
            return self.guessNumberExe(zj + 1,end)
        elif result == 0:
            return zj
        else:
            return self.guessNumberExe(head,zj-1)